src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -359,9 +404,26 @@ private static ImmutableList<ItemData>.Builder ComputeItems(LazyItemList lazyIte
                     currentList._memoizedOperation.Apply(items, currentGlobsToIgnore);
                 }
 
+                ProcessNonWildCardItemUpdates(itemsWithNoWildcards, items);
```

Comment:
Is this call necessary?

Reply:
Definitely. If the last operation were an update operation without wildcards, omitting this would make it have no effect.

Reply:
I'd add a comment saying as much.

Reply:
Emphasizing this.